src/qt/optionsmodel.cpp

```diff
@@ -315,7 +315,11 @@ bool OptionsModel::setData(const QModelIndex & index, const QVariant & value, in
             break;
         case MapPortUPnP: // core option - can be changed on-the-fly
             settings.setValue("fUseUPnP", value.toBool());
-            MapPort(value.toBool());
+            if (value.toBool()) {
+                StartMapPort();
+            } else {
+                StopMapPort();
```

Comment:
Probably call InterruptMapPort here too, before Stop, for consistency.

Edit: Oh, you have Stop call Interrupt? That's different from other [Start|Stop|Interrupt] constructions. As you probably know, the idea behind having a seperate interrupt is to have a two-stage (parallel) shutdown process, in which the interrupt can ask the subsystem to shut down, then Stop waits for it to really have stopped. Calling Stop without Interrupt could generally cause it to wait forever.

Reply:
Ok, I'll move it out.

Reply:
Sure - though if interrupt is idempotent there's no problem in calling it in Stop too.